README.md

```diff
@@ -2,6 +2,88 @@
 
 A migration tool does migrations for Konflux pipelines.
 
+Doing migrations is the major feature of pipeline-migration-tool. It auto-discovers migrations for
+given task bundle upgrades and applies found migrations to build pipelines. The pipeline can be
+either included in a `PipelineRun` definition as `spec.pipelineSpec` or a single `Pipeline` defintion.
+
+To make it work, migrations must be attached to corresponding task bundles as OCI artifacts,
+typically with `oras attach`. Let's go deep dive a bit.
+
+* If a task bundle has a migration, it must be annotated with `dev.konflux-ci.task.has-migration: true`
+* A migration must be attached to the bundle with artifact type `text/x-shellscript` and annotated
+  with `dev.konflux-ci.task.is-migration: true`.
+* A task bundle has only one migration.
+* A migration is written as a normal Bash script. Generally, it invokes `yq` to modify the pipelines.
+
+Document [Task Migration]
+(https://github.com/konflux-ci/build-definitions/?tab=readme-ov-file#task-migration) of
+konflux-ci/build-definitions describes the migrations in detailed. build-definitions provides rich
+tool chain and CI for creation, validation and ensuring migrations are available to
+pipeline-migration-tool.
+
+## Commands
+
+Pipeline migration tool supports several sub-commands to maintain the Konflux build pipelines.
+`migrate` is the main one to do migrations. Following are a few examples of the commands usage. For
+detailed information, refer to the help messages by running each one with `-h`.
+
+### To apply migrations:
+
+```bash
+cd path/to/repo
+pipeline-migration-tool migrate -u '<upgrades>'
+```
+
+`upgrades` is a JSON string encoded from a list of mappings. Each mapping includes data for a
+single task bundle upgrade, for example:
```

Comment:
Is there some schema in the renovate docs that we could link to?

Or could we at least explain that these are Renovate upgrades?